src/mbgl/renderer/layers/render_location_component_layer.cpp

```diff
@@ -0,0 +1,895 @@
+        staticImmutableCast<LocationComponentLayer::Impl>(baseImpl), unevaluated.evaluate(parameters));
+    const auto& evaluated = properties->evaluated;
+    auto& layout = impl(baseImpl).layout;
+
+    properties->renderPasses = mbgl::underlying_type(passes);
+
+    {
+        // paint
+        host->parameters.errorRadiusColor = evaluated.get<style::AccuracyRadiusColor>();
+        host->parameters.errorRadiusBorderColor = evaluated.get<style::AccuracyRadiusBorderColor>();
+
+        // layout
+        if (!layout.get<style::Bearing>().isUndefined())
+            host->parameters.puckBearing = layout.get<style::Bearing>().asConstant();
+        if (!layout.get<style::AccuracyRadius>().isUndefined())
+            host->parameters.errorRadiusMeters = layout.get<style::AccuracyRadius>().asConstant();
+        if (!layout.get<style::BearingImageSize>().isUndefined())
+            host->parameters.puckSizePx = layout.get<style::BearingImageSize>().asConstant();
+        if (!layout.get<style::TopImageSize>().isUndefined())
+            host->parameters.puckHatSizePx = layout.get<style::TopImageSize>().asConstant();
+        if (!layout.get<style::ShadowImageSize>().isUndefined())
+            host->parameters.puckShadowSizePx = layout.get<style::ShadowImageSize>().asConstant();
+        if (!layout.get<style::ImageTiltDisplacement>().isUndefined())
+            host->parameters.puckLayersDisplacement = layout.get<style::ImageTiltDisplacement>().asConstant();
+        if (!layout.get<style::PerspectiveCompensation>().isUndefined())
+            host->parameters.perspectiveCompensation = layout.get<style::PerspectiveCompensation>().asConstant();
+        host->parameters.useProperProjection = true; // <-- ToDo Fix this mode before turning it into a property.
+
+        if (!layout.get<style::Location>().isUndefined()) {
+            const std::array<double, 3> pos = layout.get<style::Location>().asConstant();
+            host->parameters.puckPosition = LatLng{pos[0], pos[1]};
+        }
+        if (!layout.get<style::BearingImage>().isUndefined())
+            host->parameters.puckImagePath = layout.get<style::BearingImage>().asConstant().id();
+        if (!layout.get<style::ShadowImage>().isUndefined())
+            host->parameters.puckShadowImagePath = layout.get<style::ShadowImage>().asConstant().id();
+        if (!layout.get<style::TopImage>().isUndefined())
+            host->parameters.puckHatImagePath = layout.get<style::TopImage>().asConstant().id();
+    }
+    evaluatedProperties = std::move(properties);
+}
+
+bool RenderLocationComponentLayer::hasTransition() const {
+    return false;
+}
+bool RenderLocationComponentLayer::hasCrossfade() const {
+    return false;
+}
+
+void RenderLocationComponentLayer::markContextDestroyed() {
+    contextDestroyed = true;
+}
+
+void RenderLocationComponentLayer::prepare(const LayerPrepareParameters& p) {
+    host->parameters.imageManager = &p.imageManager;
```

Comment:
don't keep pointer to imageManager. Lazy initialize textures when image manager is provided.